SU2_CFD/include/iteration/CIterationFactory.hpp

```diff
@@ -0,0 +1,49 @@
+/*!
+ * \file CIterationFactory.hpp
+ * \brief Headers of the iteration classes used by SU2_CFD.
+ *        Each CIteration class represents an available physics package.
+ * \author F. Palacios, T. Economon
+ * \version 7.0.5 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2020, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "CIteration.hpp"
+
+class CIterationFactory {
+
+public:
+
+  /*!
+   * \brief Deleted constructor to avoid creating instances of this class
+   */
+  CIterationFactory() = delete;
+
+  /*!
+   * \brief Create a new integration instance based on the current sub solver
+   * \param[in] integrationType  - The integration type
+   * \return                     - Pointer to the allocated integration instance
+   */
+  static CIteration* createIteration(ENUM_MAIN_SOLVER kindSolver, const CConfig *config);
```

Comment:
Does this being a class buy you anything extra? Might just make it a free function. Also, I'd capitalize Create

Reply:
I think it is more readable on the caller side if it is in a class. Of course a namespace would be also possible. I think it doesn't really make a difference here (unless you have a strong preference). Gonna change the comments and capitalize the name.

Reply:
Class gives you private methods... which you can also emulate with anonymous namespaces same difference in the end.
@talbring is you capitalize this factory then do so for all others please.

Reply:
Fine with either - you decide in the end